### PR TITLE
Add uploader of blobs

### PR DESCRIPTION
**Pull request by katcipis**
Extracting some code that would be duplicated when handling upload of blobs to azure. Uploading blobs on azure is astonishingly complicated =/ (account keys for example).

### CODE REVIEW

**Comment by katcipis** [comment]
damn travis and its 50 minute timeout =/

File: azure/blob/uploader.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import klb/azure/group
+import klb/azure/storage
+
+# Creates a new azure blob uploader
+# It will create resgroup/account/container as necessary if they do not exist
+fn azure_blob_uploader_new(resgroup, location, accountname, sku, tier, containername) {
+	# WHY: for now creating a group that already exists is ok
+	# and the function does not return any value, it will abort =/
+	# changing it now may break other people.
+	azure_group_create($resgroup, $location)
```
Comment (tiago4orion) [inline azure/blob/uploader.sh]: :/ :+1:

File: azure/blob/fs.sh [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+# and it seems more intuitive than the cp -r behavior of creating only the
+# base dir of the source path at the target path if it does not exists
+# and to copy only the contents you need the hack of regex expansion:
+# cp -r /srcdir/* /targetdir.
+#
+# This function returns an error string if it fails and "" if it succeeds.
+fn azure_blob_fs_upload_dir(fs, remotedir, localdir) {
+	# WHY: Make code handling results uniform (no relative path handling)
+	localdir <= realpath $localdir
+	out, status <= tree -if --noreport $localdir
```
Comment (tiago4orion) [inline azure/blob/fs.sh]: `tree` isn't a standard unix command. What's the problem with `find`?

File: azure/blob/fs.sh [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+	all <= split($out, "\n")
+	files = ()
+	for a in $all {
+		_, status <= test -f $a
+		if $status == "0" {
+			files <= append($files, $a)
+		}
+	}
+
+	filesprefix <= format("s:^%s::", $localdir)
```
Comment (tiago4orion) [inline azure/blob/fs.sh]: this line bugged me for a while... I thought it was some Go obscure syntax haha
Maybe rename the variable to `prefixRegex` or something related to `pattern`?

File: azure/blob/fs.sh (hunk 2 above)
Comment (katcipis) [reply azure/blob/fs.sh]: No problem, using tree was the first idea that came to my mind and I'm ignorant to what is standard unix and what is not =P. Will change to find and see if it works ok

Comment (katcipis) [reply azure/blob/fs.sh]: OMG, it is MUCH simpler with find, feeling shame =(

Comment (tiago4orion) [reply azure/blob/fs.sh]: kkkk :)

**Comment by tiago4orion** [comment]
Historical fact: In the first versions of UNIX, the filesystem was flat but organized with directories, that were just regular files that users could not write, to represent the tree layout. As the unique way to see the files were reading the directory file, the file system looked like a tree.

Then, at that time, you could have used just 'cat' to list files in a directory.

Something like:
```sh
$ cd /home/i4k
$ cat go
src
pkg
bin
$ cat go/src
github.com
$ cd go
$ cat .
src
pkg
bin
```

**Comment by katcipis** [comment]
@tiago4orion When I was thinking on ways to simulate the dirs on BLOB storage I thought about that =P, it is a pretty cool/simple idea. But it seemed excessive. Without it we have some oddities but azure already is taking too much of my time =(

**Pull request merged**